### PR TITLE
Fix base path for RDF parsing

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Fixes issue #476.

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
cc @csarven for testing -- let me know if this passes the LDN test.

**Comment by csarven** [comment]
The URIs in the output looks good to me.

**Comment by csarven** [comment]
Passes LDN tests as well! Note that it only tests JSON-LD.

However, I'm noticing that if a resource with JSON-LD uses `"@id": ""` (as the subject of triples), and then `'Accept: text/turtle'` is requested on that resource, the Turtle serialization doesn't return that triple. If the `"@id": "http://example.net/note#foo"`, it serializes to Turtle ok.

**Comment by dmitrizagidulin** [comment]
@csarven - can you open an issue for that (the turtle serialization thing you mention above), with some more details (an example file so I can reproduce it, etc)?

**Comment by dmitrizagidulin** [comment]
Is `"@id": ""` a valid JSON-LD statement?

**Comment by csarven** [comment]
Created issue https://github.com/solid/node-solid-server/issues/497

**Comment by csarven** [comment]
Yes, `"@id": ""` is valid JSON-LD. It is equivalent to `<>` in Turtle.

File: lib/utils.js [hunk 1]
```diff
@@ -65,8 +66,22 @@ function uriAbs (req) {
...
+ *   uriBase(req)  // -> 'https://example.com/files/'
+ *   ```
+ *
+ * @param req {IncomingMessage}
+ *
+ * @return {string}
+ */
 function uriBase (req) {
-  return uriAbs(req) + (req.baseUrl || '')
+  return uriAbs(req) + url.resolve(req.baseUrl, req.path)
```
Comment (RubenVerborgh) [inline lib/utils.js]: Took me some figuring out, but it seems correct indeed.

I just wonder whether `uriBase` is the right name; something like `fullUri` seems to better convey the meaning.

Comment (RubenVerborgh) [inline lib/utils.js]: Similarly, `uriAbs` doesn't seem intuitive; I actually think this one should be called `uriBase`.

**Comment by RubenVerborgh** [comment]
Had a go at the renaming in 8aea09ebd05e01af4dd2ab1c9dafb8ca32be5ec3.

**Comment by dmitrizagidulin** [comment]
@RubenVerborgh nice! :) 👍

**Pull request merged**